pathways/consignments.py

```diff
@@ -526,9 +534,19 @@ def add_contaminant_clusters_to_boxes(config, consignment):
         cluster_indexes = np.arange(
             start=cluster_start, stop=cluster_start + cluster_size
         )
-        for cluster_index in cluster_indexes:
+        for i, cluster_index in enumerate(cluster_indexes):
+            if index == len(cluster_sizes) - 1:
+                if i == len(cluster_indexes) - 1:
```

Comment:
The two if statements would be better written as:

```
            if index == len(cluster_sizes) - 1 and if i == len(cluster_indexes) - 1:
```

I'm not sure if applying the "don't iterate over last item" trick from above would be good here since the special case for the inner loops happens only at the end of the outer loop.

I have a hard time understanding the relation to clusters here. Can you please add a comment saying why this is needed? That may better document the intention than one or the other way to write this.